### PR TITLE
Migrate to zio-logging and refactor

### PR DESCRIPTION
**Pull request by mlangc**
This migrates everything to zio-logging with log4j2 and introduces a bunch of environment related refactorings, that in my opinion make the code clearer.

### CODE REVIEW

File: src/main/scala/com/schuwalow/todo/Layers.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.schuwalow.todo.config.AppConfigProvider
+import com.schuwalow.todo.config.ConfigProvider
+import com.schuwalow.todo.config.DbConfigProvider
+import com.schuwalow.todo.repository.DoobieTodoRepository
+import com.schuwalow.todo.repository.TodoRepository
+
+object Layers {
+
+  type Layer0Env =
```
Comment (mschuwalow) [inline src/main/scala/com/schuwalow/todo/Layers.scala:16]: I like that organization 👍

File: src/main/scala/com/schuwalow/todo/repository/DoobieTodoRepository.scala [hunk 2]
```diff
@@ -26,7 +27,7 @@ final class DoobieTodoRepository(xa: Transactor[Task]) {
 
   val todoRepository = new TodoRepository.Service {
 
-    def getAll(): UIO[List[TodoItem]] =
+    def getAll: UIO[List[TodoItem]] =
```
Comment (mschuwalow) [inline src/main/scala/com/schuwalow/todo/repository/DoobieTodoRepository.scala:30]: nice cleanup.
these could be val as well

File: src/main/scala/com/schuwalow/todo/repository/TodoRepository.scala [hunk 3]
```diff
@@ -28,4 +30,40 @@ object TodoRepository extends Serializable {
       todoItemForm: TodoItemPatchForm
     ): UIO[Option[TodoItem]]
   }
+
+  def withTracing[RIn, ROut <: TodoRepository with Logging, E](
```
Comment (mschuwalow) [inline src/main/scala/com/schuwalow/todo/repository/TodoRepository.scala:34]: nice

File: src/main/scala/com/schuwalow/todo/repository/DoobieTodoRepository.scala (hunk 2 above)
Comment (mlangc) [reply src/main/scala/com/schuwalow/todo/repository/DoobieTodoRepository.scala:30]: I've just changed them to vals

**Comment by mlangc** [comment]
Thanks for the review! As far as I'm concerned this can be merged.

File: src/main/scala/com/schuwalow/todo/Layers.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  type AppEnv = Layer2Env
+
+  object live {
+
+    val layer0: ZLayer[Blocking, Throwable, Layer0Env] =
+      Blocking.any ++ ConfigProvider.live ++ Slf4jLogger.make((_, msg) => msg)
+
+    val layer1: ZLayer[Layer0Env, Throwable, Layer1Env] =
+      AppConfigProvider.fromConfig ++ DbConfigProvider.fromConfig ++ ZLayer.identity
```
Comment (hmemcpy) [inline src/main/scala/com/schuwalow/todo/Layers.scala:33]: Hey, awesome work! I Just have a question: what is the purpose of `ZLayer.identity`?

Comment (mlangc) [reply src/main/scala/com/schuwalow/todo/Layers.scala:33]: Thank you :-). The `identity` just passes on the previous layer unmodified (remember that we have `type LayerN = PreviousLayer ++ Modules`).

Comment (hmemcpy) [reply src/main/scala/com/schuwalow/todo/Layers.scala:33]: @mlangc Awesome, makes sense :)

**Comment by mschuwalow** [comment]
merging this.
Thanks for your work!

**Pull request merged**